### PR TITLE
docs: Update link to Javadoc

### PR DESCRIPTION
**Pull request by arriolac**
Now that we are uploading the `*-javadoc.jar` file, docs are automatically viewable at javadoc.io.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,8 +52,7 @@ Or, if you'd like to contribute, send us a [pull request] and refer to our [code
 You can also reach us on our [Discord channel].
 
 For more information, check out the detailed guide on the
-[Google Developers site][devsite-guide]. You can also view the generated
-[reference docs][javadoc] for a full list of classes and their methods.
```
Comment (barbeau) [inline README.md]: I don't think `[javadoc]` was intended to be removed here?

Comment (arriolac) [reply README.md]: That was intentional since the text is in more prominent in its own section now. We can add that back if you think it'll help?

**Comment by arriolac** [comment]
> A comment in-line.
> 
> Also, I don't see v1.0.2 docs at https://www.javadoc.io/doc/com.google.maps.android/android-maps-utils/latest/index.html? Would it still be propagating?

Hmm... odd. I had to visit this link first:

https://www.javadoc.io/doc/com.google.maps.android/android-maps-utils/1.0.2/index.html

and now it's visible at:

https://www.javadoc.io/doc/com.google.maps.android/android-maps-utils/latest/index.html

File: README.md (hunk 1 above)
Comment (barbeau) [reply README.md]: Oh, sorry - somehow I missed that looking at the diff, LGTM.

**Pull request merged**